File: IN1010/trix/go/enkelarrayliste/enkelarrayliste.go
package main

type iterator struct {
	array []string
	index int
}

func NewIterator(a []string) *iterator {
	return &iterator{
		array: a,
		index: 0,
	}
}

func (i *iterator) HasNext() bool {
	return i.index < len(i.array)
}

func (i *iterator) Next() string {
	if !i.HasNext() {
		panic("No more elements")
	}

	ret := i.array[i.index]
	i.index++
	return ret
}

File: IN1010/trix/07/iterator/EnkelArrayListe.java
import java.util.Iterator;

class EnkelArrayListe implements Iterator<String> {
    String[] array;
    private int i = 0;

    public EnkelArrayListe(int n) {
        array = new String[n];
    }

    public void LeggInn(String s) {
        for (int i = 0; i < array.length; i++) {
            if (array[i] == null) {
                array[i] = s;
                return;
            }
        }
    }

    @Override
    public boolean hasNext() {
        if (this.i < this.array.length) {
            return true;
        }
        return false;
    }

    @Override
    public String next() {
        if (this.hasNext()) {
            String ret = array[this.i];
            this.i++;
            return ret;
        }
        return "error";
    }


}
